src/resource/saveSolidDatasetAt.ts

```diff
@@ -47,7 +50,7 @@ import type { FetchOptions } from "../type/FetchOptions";
 export async function saveSolidDatasetAt<Dataset extends SolidDataset>(
   datasetUrl: UrlString,
   solidDataset: Dataset,
-  accessGrant: VerifiableCredential,
+  accessGrant: VerifiableCredential | DatasetWithId,
```

Comment:
In other instances in this PR, `VerifiableCredential` has been replaced by `DatasetWithId`. Why the union here?

Reply:
The `VerifiableCredential` is not necessary from a type safety perspective as it is a superset of `DatasetWithId` (as are `AccessGrant` and `AccessRequest`). These are only left in the type union to make it more readable for humans looking at the types.